queue_job/jobrunner/runner.py

```diff
@@ -219,16 +219,14 @@ def close(self):
 
     def _has_queue_job(self):
         with closing(self.conn.cursor()) as cr:
-            try:
-                cr.execute("SELECT 1 FROM ir_module_module "
-                           "WHERE name=%s AND state=%s",
-                           ('queue_job', 'installed'))
-            except psycopg2.ProgrammingError as err:
-                if unicode(err).startswith('relation "ir_module_module" '
-                                           'does not exist'):
-                    return False
-                else:
-                    raise
+            cr.execute("SELECT to_regclass('ir_module_module')")
+            [ir_module_module_exists] = cr.fetchone()
+            if not ir_module_module_exists:
+                return False
+            cr.execute(
+                "SELECT 1 FROM ir_module_module WHERE name=%s AND state=%s",
+                ('queue_job', 'installed')
+            )
```

Comment:
Alternatively, we could check postgresql's error code, it's 42P01 for an undefined table. (https://www.postgresql.org/docs/8.1/static/errcodes-appendix.html). It's one query less, but also one error more in the logs. 

`to_regclass` is only available since PG9.4 though and I think lesser versions should be supported in Odoo 10.0